clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/VeniceAfterImageConsumerImpl.java

```
@@ -191,24 +189,22 @@ public void run() {
         // Seek to end of push
         if (currentVersion != maxVersion) {
           // get current subscriptions and seek to endOfPush
-          Set<Integer> partitions = new HashSet<>();
           for (PubSubTopicPartition partitionSubscription: subscriptions) {
             partitions.add(partitionSubscription.getPartitionNumber());
           }
-          try {
-            LOGGER.info(
-                "New Version detected!  Seeking consumer to version: " + currentVersion
-                    + " in VersionSwaDetectionThread: " + this.getId());
-            seekToEndOfPush(partitions).get();
-          } catch (InterruptedException | ExecutionException e) {
-            LOGGER.error(
-                "Seek to End of Push Failed for store: " + storeName + " partitions: " + partitions
-                    + " on VersionSwapDetectionThread: " + this.getId() + "will retry...",
-                e);
-          }
+        }
+        try {
+          LOGGER.info(
+              "New Version detected!  Seeking consumer to version: " + currentVersion + " in consumer: "
+                  + changelogClientConfig.getConsumerName());
+          seekToEndOfPush(partitions).get();
+        } catch (InterruptedException | ExecutionException e) {
```

can we catch all Exception here?

---

Yeah ok

---

Also a bigger try-catch.

---

ok